bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/ProfileCallback.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.thing.profiles;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * Gives access to the framework features for continuing the communication flow.
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ *
+ */
+@NonNullByDefault
+public interface ProfileCallback {
+
+    /**
+     * Forward the given command to the respective thing handler.
+     *
+     * @param command
+     */
+    void handleCommand(Command command);
+
+    /**
+     * Forward the given state update to the respective thing handler.
+     *
+     * @param state
+     */
+    void handleUpdate(State state);
+
+    /**
+     * Send a command to the framework.
+     *
+     * @param command
+     */
+    void sendCommandEvent(Command command);
```

Comment:
I would remove the word "Event" here - after all, you are sending a command, not an event.